### PR TITLE
change requires to install_requires in setup.py and add sphinx-galler…

### PR DESCRIPTION
**Pull request by mehdidc**
…y to  requirements

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -88,7 +88,8 @@
               'py_modules': ['pyearth.earth', 'pyearth._version'],
               'ext_modules': ext_modules,
               'classifiers': ['Development Status :: 3 - Alpha'],
-              'requires': ['numpy', 'scipy', 'sklearn']}
+              'install_requires': ['numpy', 'scipy', 'scikit-learn',
+                                   'sphinx_gallery']}
```
Comment (agramfort) [inline setup.py]: requiring numpy and scipy install with pip is dangerous. I've had terrible experience
with automatic update of my numpy setup with pip. I'd keep numpy / scipy as requires
and not automatically installed if possible.

**Comment by mehdidc** [comment]
Thanks @agramfort  for the suggestion, changed.

**Comment by agramfort** [comment]
LGTM

**Comment by jcrudy** [comment]
Looks good.  Thanks!

**Pull request merged**